src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -56,23 +60,35 @@ public void Initialize()
         /// <summary>Continuously requests matured blocks from another chain.</summary>
         private async Task RequestMaturedBlocksContinouslyAsync()
         {
-            while (!this.cancellation.IsCancellationRequested)
+            try
             {
-                bool delayRequired = await this.AskForBlocksAsync().ConfigureAwait(false);
+                // Initialization delay.
+                // Give other node some time to start API service.
+                await Task.Delay(InitializationDelayMs, this.cancellation.Token).ConfigureAwait(false);
 
-                if (delayRequired)
+                while (!this.cancellation.IsCancellationRequested)
                 {
-                    // Since we are synced or had a problem syncing there is no need to ask for more blocks right away.
-                    // Therefore awaiting for a delay during which new block might be accepted on the alternative chain
-                    // or alt chain node might be started.
-                    await Task.Delay(RefreshDelayMs).ConfigureAwait(false);
+                    bool delayRequired = await this.AskForBlocksAsync(this.cancellation.Token).ConfigureAwait(false);
```

Comment:
I find the naming of this quite unsettling.
We ask for block and then receive a bool that says whether a delay is required.
Can we maybe change this to something more obvious?

Reply:
Sure. Renamed